pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,490 @@
+			},
+			workloadCmpOpts: defaultWorkloadCmpOpts,
+		},
+		"pod is finalized when it's succeeded": {
+			pod: *basePodWrapper.
+				Clone().
+				Label("kueue.x-k8s.io/managed", "true").
+				KueueFinalizer().
+				StatusPhase(corev1.PodSucceeded).
+				Obj(),
+			wantPod: basePodWrapper.
+				Clone().
+				Label("kueue.x-k8s.io/managed", "true").
+				StatusPhase(corev1.PodSucceeded).
+				Obj(),
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+					Admitted(true).
+					Obj(),
+			},
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+					Admitted(true).
+					Condition(metav1.Condition{
+						Type:    "Finished",
+						Status:  "True",
+						Reason:  "JobFinished",
+						Message: "Job finished successfully",
+					}).
+					Obj(),
+			},
+			workloadCmpOpts: append(
+				defaultWorkloadCmpOpts,
+				// This is required because SSA doesn't work properly for the fake k8s client.
+				// Reconciler appends "Finished" condition using SSA. Fake client will just
+				// replace all the older conditions.
+				// See: https://github.com/kubernetes-sigs/controller-runtime/issues/2341
+				cmpopts.IgnoreSliceElements(func(c metav1.Condition) bool {
+					return c.Type == "Admitted"
+				}),
+			),
+		},
+		"pod without scheduling gate is terminated if workload is not admitted": {
+			initObjects: []client.Object{
+				utiltesting.MakeResourceFlavor("unit-test-flavor").Label("kubernetes.io/arch", "arm64").Obj(),
+			},
+			pod: *basePodWrapper.
+				Clone().
+				Label("kueue.x-k8s.io/managed", "true").
+				KueueFinalizer().
+				Obj(),
+			wantPod: basePodWrapper.
+				Clone().
+				Label("kueue.x-k8s.io/managed", "true").
+				KueueFinalizer().
+				StatusConditions(corev1.PodCondition{
+					Type:    "TerminationTarget",
+					Status:  corev1.ConditionTrue,
+					Reason:  "StoppedByKueue",
+					Message: "Not admitted by cluster queue",
+				}).
+				Obj(),
+			workloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					Obj(),
+			},
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("unit-test", "ns").Finalizers(kueue.ResourceInUseFinalizerName).
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 1).Request(corev1.ResourceCPU, "1").Obj()).
+					Obj(),
+			},
+			workloadCmpOpts: defaultWorkloadCmpOpts,
+		},
```

Comment:
add a test for finished pod without a workload.

Reply:
Done in 4e659280